Gems/ROS2/Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "LidarRaycaster.h"
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzFramework/Physics/Common/PhysicsSceneQueries.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <AzFramework/Physics/PhysicsSystem.h>
+#include <AzFramework/Physics/Shape.h>
+
+namespace ROS2
+{
+    void LidarRaycaster::SetRaycasterScene(const AzPhysics::SceneHandle& handle)
+    {
+        m_sceneHandle = handle;
+    }
+
+    AZStd::vector<AZ::Vector3> LidarRaycaster::PerformRaycast(
+        const AZ::Vector3& start,
+        const AZStd::vector<AZ::Vector3>& directions,
+        const AZ::Transform& globalToLidarTM,
+        float distance,
+        bool ignoreLayer,
+        unsigned int ignoredLayerIndex) const
+    {
+        AZStd::vector<AZ::Vector3> results;
+        if (m_sceneHandle == AzPhysics::InvalidSceneHandle)
+        {
+            AZ_Warning("LidarRaycaster", false, "No valid scene handle");
+            return results;
+        }
+
+        AzPhysics::SceneQueryRequests requests;
+        requests.reserve(directions.size());
+        results.reserve(directions.size());
+        for (const AZ::Vector3& direction : directions)
+        { // NOTE - performance-wise, consider reusing requests
```

Comment:
This almost looks like a TODO. Probably no need for this comment

Reply:
yes, it can go away. I have also checked that this does not provide a notable improvement.

Reply:
Done